src/QSymbolicsBase/QSymbolicsBase.jl

```diff
@@ -137,25 +138,14 @@ const QObj = Union{AbstractBra,AbstractKet,AbstractOperator,AbstractSuperOperato
 const SymQObj = Symbolic{<:QObj} # TODO Should we use Sym or Symbolic... Sym has a lot of predefined goodies, including metadata support
 Base.:(-)(x::SymQObj) = (-1)*x
 Base.:(-)(x::SymQObj,y::SymQObj) = x + (-y)
+Base.hash(x::SymQObj, h::UInt) = isexpr(x) ? hash(arguments(x), h) : hash([typeof(x),basis(x)], h)
 
-function _in(x::SymQObj, y::SymQObj)
-    for i in arguments(y)
-        if isequal(x, i)
-            return true
-        end
-    end
-    false
-end
 function Base.isequal(x::X,y::Y) where {X<:Union{SymQObj, Symbolic{Complex}}, Y<:Union{SymQObj, Symbolic{Complex}}}
```

Comment:
I did not notice this in the previous PR, but I think this causes piracy because it will be the method to use when `isequal(::Symbolic{Complex},::Symbolic{Complex})` is called and we do not own neither `Symbolic` nor `Complex`.

To avoid that piracy (which get's detected by the `Aqua` test but it is currently turned off because there are a lot of other Piracies too), I suggest this gets renamed to `_isequal` without any type constraints and then two separate `Base.isequal` get created:

`Base.isequal(x::Union{...}, y::SymQObj) = _isequal(x, y)`
`Base.isequal(x::SymQObj, y::::Union{...}) = _isequal(x, y)`

That way you do not risk dispatching on this if you have two `Complex` arguments.

Reply:
Which we also want to avoid because this will anyway fail if given two sums of complex scalars, because the complex sum probably does not have `_set_precomputed`

Reply:
I might be misunderstanding you, but won't your approach of creating two separate `Base.isequal` methods create ambiguity? For instance, if you are simply evaluating equality of a symbolic operator, then it's unclear which specific method to use. Just to clarify, when you say `Union{...}` you mean `Union{SymQObj, Symbolic{Complex}`, right?

Reply:
You are right, that would create an ambiguity, this was a bad suggestion.

Better suggestion: just remove the Complex from this one and define `isequal(::SymQObj, ::Symbolic{Complex})=false` and the other order of arguments

Reply:
I agree with this, but this still doesn't help when we're defining equality between inner products. How about we additionally define `Base.isequal(x::SBraKet, y::SBraKet) = isequal(x.bra, y.bra) && isequal(x.ket, y.ket)`?

Reply:
yup, that sounds necessary

There are other ones that might be necessary too, but let's not worry about them until they pop up